### ISSUE TITLE
Feature Request: Use some read only API without username and password

### ISSUE THREAD

**Issue by guoyunhe** [openSUSE/open-build-service#7510]
**Is your feature request related to a problem? Please describe.**

I made a tool [opi](https://github.com/openSUSE-zh/opi) to help users search packages of OBS from comnand line.

However, to use any OBS API, username and password are required.

1. If I hard code username and password, it is easier to use but the password will be exposed. [Code](https://github.com/openSUSE-zh/opi/blob/master/opi#L37)
2. If I let users sign in with their own username and password, it will be very hard to use. Many openSUSE users don't even have a username.

The only API I need is the binary search API. 

**Describe the solution you'd like**

I read the API of GitHub. Some of its API (read only, no private/sensitive data) don't require secrets, tokens, passwords. [See here](https://developer.github.com/v3/#parameters)

```
curl -i "https://api.github.com/repos/vmg/redcarpet/issues?state=closed"
```

For OBS, some APIs, like binary search, file download, could be public and require no username and password. It will be easier for us to use the API in tools and avoid expose password to hackers.

**Describe alternatives you've considered**
An alternative way is to only require username, not password.

**Additional context**

**Comment by hennevogel** [comment]
You will also find that github and most other public apis do some form of aggressive [rate limiting](https://developer.github.com/v3/#rate-limiting). The reason we don't have a public API is that we don't have rate limiting currently.

**Comment by guoyunhe** [comment]
Indeed, rate limiting is a must-have feature if you have public API.
